js/components/screenGroupDetails.js

```diff
@@ -17,20 +40,33 @@ window.ScreenGroupDetails = React.createClass({
     }
     return name;
   },
+  renderFinishButton: function () {
+    return (
+      <a
+        className="btn btn-positive btn-block"
+        onClick={this.handleSubmitGroup}
+      >Finish</a>
+    );
+  },
   renderGroupRules: function () {
     var groupId = this.props.groupId;
-    var rules;
     var i;
     var groupJson = this.props.getGroups();
     for (i = 0; i < groupJson.length; i++) {
       if (groupJson[i].id === groupId) {
-        rules = groupJson[i].groupRules;
+        this.setState({
+          rules: groupJson[i].groupRules
```

Comment:
I think I'm wrong..I did something like this
`componentDidMount:function () {`
`var groupId = this.props.groupId;`
`var i;`
`var groupJson = this.props.getGroups();`
`for (i = 0; i < groupJson.length; i++) {`
`if (groupJson[i].id === groupId) {`
`this.setState({`
`rules: groupJson[i].groupRules`
`});`
`}`
`}`
`},`

The llint shows the error 

> Donot use state in `componentDidMount`